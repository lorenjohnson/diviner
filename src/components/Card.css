.card {
  margin: 1em;
  border-radius: 16px;
  background-color: oldlace;
  box-shadow: 0px 2px 6px 0px rgba(245, 243, 243, 0.61);
  cursor: pointer;
}

.card img {
  padding: 1em;
  /* Fixing height/width as proportions of the front
  and back images are not exaclty the same */
  height: calc(17.3 * 1.6rem);
  width: calc(10 * 1.6rem);
}

/* Display card back when face-down and front when not */

.card.face-down .card-front,
.card:not(.face-down) .card-back {
  display: none;
}

.drawn-cards:hover .card {
  opacity: 0.7;
}

.drawn-cards .card:not(.face-down):hover {
  opacity: 1;
  box-shadow: 3px 3px 0px 0px rgba(245, 243, 243, 0.61);
}
