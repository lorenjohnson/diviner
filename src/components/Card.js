import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import './Card.css'

export default class Card extends Component {
  static propTypes = {
    cardId: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number
    ]),
    frontImageURL: PropTypes.string.isRequired,
    backImageURL: PropTypes.string.isRequired,
    name: PropTypes.string,
    className: PropTypes.string,
    onClick: PropTypes.func,
    onMouseOver: PropTypes.func,
    onMouseOut: PropTypes.func
  }

  state = {
    faceDown: true
  }

  flipOver = () => this.setState(state => ({ faceDown: !state.faceDown }))

  onClick = () => this.props.onClick
    ? this.props.onClick(this.props.cardId)
    : this.flipOver(this.state.faceDown)

  onMouseOver = () => this.props.onMouseOver && !this.state.faceDown
    ? this.props.onMouseOver(this.props.cardId)
    : undefined

  onMouseOut = () => this.props.onMouseOut
    ? this.props.onMouseOut(this.props.cardId)
    : undefined

  render () {
    const { frontImageURL, backImageURL, reversed, className } = this.props
    const { faceDown } = this.state

    return (
      <div className={classNames(className, 'card', { 'face-down': faceDown, reversed })}
        onClick={this.onClick}
        onMouseOver={this.onMouseOver}
        onMouseOut={this.onMouseOut}>
        <CardImage className='card-back' imageURL={backImageURL} />
        <CardImage className='card-front' imageURL={frontImageURL} />
      </div>
    )
  }
}

export function CardImage ({ imageURL, className }) {
  return <div className={className}>
    <img src={imageURL} alt={imageURL} />
  </div>
}
