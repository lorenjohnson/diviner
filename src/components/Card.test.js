import React from 'react'
import ReactTestRenderer from 'react-test-renderer'
import Card from './Card'

it('matches last snapshot', () => {
  const component = ReactTestRenderer.create(
    <Card frontImageURL='/frontimage.jpg' backImageURL='/backimage.jpg' />
  )
  expect(component.toJSON()).toMatchSnapshot()
})

// test whether cardId is sent to each call back methods
// flip over state (className check?)
// CardImage test? or move to it's own file?