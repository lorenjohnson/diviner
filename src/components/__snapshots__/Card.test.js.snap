// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches last snapshot 1`] = `
<div
  className="card"
  onClick={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="card-back"
  >
    <img
      alt="/backimage.jpg"
      src="/backimage.jpg"
    />
  </div>
  <div
    className="card-front"
  >
    <img
      alt="/frontimage.jpg"
      src="/frontimage.jpg"
    />
  </div>
</div>
`;
